Python/chapter01/1.3 - URLify/miguel_1.3_sol.py

```diff
@@ -0,0 +1,96 @@
+"""
+Python version 3.7.0
+1.3 - URLify
+Write a method to replace all spaces in a string with "%20".
+The following only applies to C/C++: "You may assume that the string has sufficient
+space at the end to hold the additional characters, and that you are given the "true" length of the string"
+(Note: If implementing in Java, please use a character array so that you can perform this operation in place)
+Since we are using python, no need to use true length.  I will have two functions, one with
+true_length and the other without.
+"""
+import unittest
+
+
+def urlify(s: str, true_length: int) -> str:
+    """
+    Given a string and it's "true" length, this function will return a new string
+    that replaces all of the spaces of the input string with '%20'.
+    Precondition(s):
+    - length of s <= true_length
+    :param s: the original string to 'urlify'
+    :param true_length: since s may have additional characters, we focus on true_length instead of actual s length
+    :return: string with each space from s replaced with '%20'
+    """
+    # Below link goes over string concat efficiency in python (I will use method 4
+    # https://waymoot.org/home/python_string/
+    # each space will make us replace it (with '%', then add 2 characters ('2', '0')
+    result = ['\x00'] * len(s)*2  # extra buffer space
+    i = 0
+    space_count = 0
+    added_chars_per_space = 2
+    for c in s:
```

Comment:
i forgot to add a restriction to the challenge: you cannot reference `s`: in C/C++, you wouldn't have that luxury.